dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h12/http2/GenericHttp2ServerTransportListener.java

```diff
@@ -187,6 +187,12 @@
         return serverChannelObserver;
     }
 
+    @Override
+    public void onStreamClosed() {
+        // doing on event loop thread
+        getStreamingDecoder().close();
```

Comment:
**Performance Issue**: Closing the streaming decoder on stream closure without checking if all data has been processed might lead to data loss. This is particularly important for ensuring data integrity in streaming applications. <br> **Fix**: Before closing the streaming decoder, check if all expected data has been processed. Implement a mechanism to ensure data integrity, such as acknowledgments from the client that all data has been received and processed. <br> **Code Suggestion**: 
 ```
 +    @Override
 +    public void onStreamClosed() {
 +        // Ensure all expected data has been processed before closing the streaming decoder
 +        if (!getStreamingDecoder().isDataFullyProcessed()) {
 +            // Implement mechanism to ensure data integrity, e.g., acknowledgments from the client
 +            return;
 +        }
 +        getStreamingDecoder().close();
 +    }
 ```